application/models/Permission.php

```diff
@@ -785,6 +792,23 @@ public static function getUserRole($iUserID)
         return UserInPermissionrole::model()->getRoleForUser($iUserID);
     }
 
+        /**
+     * get permission by user roles
+     * @param integer[] $rolesIds array of roles id
+     * @param string $permission;
+     * @param string $crud
+     * @return boolean
+     */
+    public static function getPermissionByRoles($rolesIds, $permission, $crud = 'read')
```

Comment:
We can really use this function in https://github.com/LimeSurvey/LimeSurvey/blob/57a22a6b6b44d913159c683598bf9a76e2dad53e/application/models/Permission.php#L607-L614

Reply:
I agree as functions so similar, but also would need to updated here: https://github.com/LimeSurvey/LimeSurvey/blob/0a9690936d48465ecc0eadf2b12933b81ae26cf3/application/models/Permissiontemplates.php#L370  and https://github.com/LimeSurvey/LimeSurvey/blob/0a9690936d48465ecc0eadf2b12933b81ae26cf3/application/views/userRole/partials/_permissions.php#L92

Reply:
This patch resolves the issue. I think could be applied as is, and then update and remove references to reoleHasPermission later. Or could be done in one patch for completeness?

Reply:
> I agree as functions so similar, but also would need to updated here:

- roleHasPermission function : one role give has permission
- getPermissionByRoles function : list of roles

You think allow  `$iRoleId` to be an array in https://github.com/LimeSurvey/LimeSurvey/blob/0a9690936d48465ecc0eadf2b12933b81ae26cf3/application/models/Permission.php#L720 ?

Alternative solution : 
````
    /**
     * get permission by user roles
     * @param integer $userId array of roles id
     * @param string $permission;
     * @param string $crud
     * @return boolean|null : boolean if have permission set, null if not
     */
    public static function getPermissionByRoles($userID, $permission, $crud = 'read')
    {
        $aRolesList = CHtml::listData(self::getUserRole($userID), 'ptid', 'ptid');
        if (empty($aRolesList)) {
            return null;
        }
        $criteria = new CDbCriteria();
        $criteria->compare("entity", "role");
        $criteria->compare("permission", $permission);
        $criteria->addInCondition('entity_id', $rolesIds);
        $criteria->compare($crud . '_p', 1);
        return boolval(self::model()->count($criteria));
    }
````
 For using it 
````
    $byRole = self::getPermissionByRoles($userID, 'superadmin')
    if (!is_null($byRole)) {
        $permission = $byRole;
    }
````

But here : usage is really more complex … if user don't have role : permission came from Permission, if have role : came from role.

2nd alt solution

- After plugin event : check if user have role
- If yes : create a new function checking only role for ALL system.

Reply:
> This patch resolves the issue. I think could be applied as is, and then update and remove references to reoleHasPermission later. Or could be done in one patch for completeness?

If possible : since i add function : it must be OK to be used elsewhere by other dev.